File: src/test/java/org/gradle/plugins/fsm/util/BaseService.java
package org.gradle.plugins.fsm.util;

import de.espirit.firstspirit.module.ServerEnvironment;
import de.espirit.firstspirit.module.Service;
import de.espirit.firstspirit.module.ServiceProxy;
import de.espirit.firstspirit.module.descriptor.ServiceDescriptor;

public class BaseService implements Service<ServerEnvironment> {
    @Override
    public void start() {

    }

    @Override
    public void stop() {

    }

    @Override
    public boolean isRunning() {
        return false;
    }

    @Override
    public Class<? extends ServerEnvironment> getServiceInterface() {
        return null;
    }

    @Override
    public Class<? extends ServiceProxy<ServerEnvironment>> getProxyClass() {
        return null;
    }

    @Override
    public void init(ServiceDescriptor serviceDescriptor, ServerEnvironment serverEnvironment) {

    }

    @Override
    public void installed() {

    }

    @Override
    public void uninstalling() {

    }

    @Override
    public void updated(String s) {

    }
}

File: src/test/kotlin/org/gradle/plugins/fsm/util/BaseService.kt
package org.gradle.plugins.fsm.util

import de.espirit.firstspirit.module.ServerEnvironment
import de.espirit.firstspirit.module.Service
import de.espirit.firstspirit.module.ServiceProxy
import de.espirit.firstspirit.module.descriptor.ServiceDescriptor

open class BaseService: BaseServiceItf, Service<BaseServiceItf> {

    override fun start() {}

    override fun stop() {}

    override fun isRunning(): Boolean {
        return false
    }

    override fun getServiceInterface(): Class<out BaseServiceItf> {
        return BaseServiceItf::class.java
    }

    override fun getProxyClass(): Class<out ServiceProxy<BaseServiceItf>>? {
        return null
    }

    override fun init(serviceDescriptor: ServiceDescriptor?, serverEnvironment: ServerEnvironment?) {}

    override fun installed() {}

    override fun uninstalling() {}

    override fun updated(s: String?) {}

}